mlflow/pyfunc/backend.py

```diff
@@ -1,39 +1,138 @@
 import logging
 import os
+import hashlib
+import shutil
+from pathlib import Path
 
+import yaml
 import subprocess
 import posixpath
 from mlflow.models import FlavorBackend
 from mlflow.models.docker_utils import _build_image, DISABLE_ENV_CREATION
 from mlflow.models.container import ENABLE_MLSERVER
 from mlflow.pyfunc import ENV, scoring_server, mlserver
+from mlflow.exceptions import MlflowException
 
-from mlflow.utils.conda import get_or_create_conda_env, get_conda_bin_executable, get_conda_command
+from mlflow.utils.conda import (
+    get_or_create_conda_env,
+    get_conda_bin_executable,
+    get_conda_command,
+)
 from mlflow.tracking.artifact_utils import _download_artifact_from_uri
 from mlflow.utils.file_utils import path_to_local_file_uri
+from mlflow.utils.environment import (
+    _PYTHON_ENV_FILE_NAME,
+    _CONDA_ENV_FILE_NAME,
+    PythonEnv,
+    EnvManager,
+    _contains_conda_packages,
+)
 from mlflow.version import VERSION
 
 _logger = logging.getLogger(__name__)
 
+_MLFLOW_HOME_ENV_VAR = "MLFLOW_HOME"
+_MLFLOW_ENV_ROOT_ENV_VAR = "MLFLOW_ENV_ROOT"
+
+
+def _is_windows():
+    return os.name == "nt"
+
+
+def _join_commands(*commands):
+    sep = " & " if _is_windows() else " && "
+    return sep.join(commands)
+
+
+def _get_entrypoint():
+    return ["cmd", "/c"] if _is_windows() else ["bash", "-c"]
+
+
+def _run_command(command, **kwargs):
+    prc = subprocess.Popen(
+        command,
+        shell=_is_windows(),
```

Comment:
For non-windows, `shell` should also be `True` otherwise I think `_run_multiple_commands` case it will fail.

And does tests cover the case ?

Reply:
> I think _run_multiple_commands case it will fail.

Are you sure? The code below is equivalent to what `_run_multiple_commands` and works.

```
import subprocess

prc = subprocess.Popen(["bash", "-c", "echo foo && echo bar"])
returncode = prc.wait()
```

output:

```
foo
bar
```

Reply:
Oh I miss the "*_get_entrypoint()" part which add "bash -c" .
you are right.

Btw, why for windows case we need add "shell=True" ? For windows we also prepend "cmd /c" to the commands

Reply:
To use `cmd`

Reply:
I actually forgot why I set shell=True for windows.

Reply:
Trying `shell=True` without `"bash -c"` or `"cmd /c"` now

Reply:
Ended up using `cmd /c` and `bash -c` with `shell=False`